hyperlink/_url.py

```diff
@@ -907,7 +888,6 @@ def replace(self, scheme=_UNSET, host=_UNSET, path=_UNSET, query=_UNSET,
             port=_optional(port, self.port),
             rooted=_optional(rooted, self.rooted),
             userinfo=_optional(userinfo, self.userinfo),
-            family=_optional(family, self.family),
```

Comment:
I do think we probably need a deprecation warning here.  This is public, documented functionality that's disappearing, and an exception is a rude way to find out about that.

That said, this is a very easy deprecation to do, because the passed value can more or less just be ignored.